source/taup/taup_wavefront.rst

```diff
@@ -0,0 +1,112 @@
+taup wavefront
+==============
+
+:本节贡献者: |田冬冬|\（作者）、
+             |姚家园|\（审稿）
+:最近更新日期: 2021-06-05
+
+----
+
+``taup wavefront`` 命令可用于生成波前面，即任意时刻地震波在地球内部的波场快照。
+
+基本用法
+--------
+
+使用 ``taup wavefront --help`` 可查询该命令的完整用法和选项。其选项与
+:doc:`taup_time` 几乎完全相同，故不再重复介绍。此处仅介绍常用用法。
+
+计算震源深度为 300 km，震中距 60° 处，P 震相的波前面::
+
+    $ taup wavefront -ph P -deg 60 -h 300
+
+该命令会生成文件 :file:`taup_wavefront.gmt`\ ，其内容如下::
+
+    > P at 100.0 seconds
+    5.31    5151.8   100.00  254.331
+    5.42    5158.7   100.00  260.031
+    5.49    5163.9   100.00  264.250
+    5.55    5167.7   100.00  267.326
+    ...(省略若干行)
+    > P at 200.0 seconds
+    15.69    4244.7   200.00  254.331
+    15.93    4267.0   200.00  260.031
+    16.10    4283.9   200.00  264.250
+    16.23    4296.4   200.00  267.326
+    ...(省略若干行)
+
+该文件中包含了不同时刻（t=100, 200, 300, ...）P 波的波前面信息。输出包含四列，
+分别是：
+
+- 震中距（度）
+- 深度（km）
+- 走时（秒）
+- 射线参数（秒/度）
+
+可以直接使用文件的前两列进行绘图。
+
+加上 ``-gmt`` 选项，则会生成一个可用于绘制波前面传播的 GMT 脚本 :file:`taup_wavefront.gmt`::
+
+    $ taup wavefront -ph P -deg 60 -h 300 -gmt
+
+使用如下命令执行该绘图脚本，即可得到 PDF 格式的波前面传播图::
+
+    $ bash taup_wavefront.gmt
+
+绘制出来的图件效果如下：
+
+.. image:: taup_wavefront.jpg
+   :width: 500 px
+   :align: center
+
+进阶用法
+--------
+
+``taup wavefront`` 提供了 ``--rays`` 和 ``--timestep`` 选项进一步控制生成的波前面时
+所使用的射线数目（即每个波前面时用多少个点约束）和时间步长（即间隔多长时间产生一个
+波前面）。
```

Comment:
```suggestion
``taup wavefront`` 提供的 ``--rays`` 选项可以进一步控制生成波前面时所使用的射线数目
（即每个波前面用多少个点约束），而 ``--timestep`` 选项可以时间步长
（即间隔多长时间产生一个波前面）。
```

Reply:
I am still not clear about `--rays` option.

Reply:
Let's say you want to manually calculate the wavefront for t=100 s. How to do that? You can set up 10 rays with different parameters, calculate the ray path of each ray and determine the traveltimes at each position along the ray path. With 10 rays, you can get 10 positions for t=100 s, and the 10 positions can be used to draw the wavefront.

Reply:
Clear.